b3sum/src/main.rs

```diff
@@ -177,8 +184,94 @@ fn read_key_from_stdin() -> Result<[u8; blake3::KEY_LEN]> {
     }
 }
 
+fn check_file(input: impl Read, mmap_disabled: bool) -> Result<u64> {
+    let hasher = blake3::Hasher::new();
+    let mut num_failed = 0;
+
+    let reader = std::io::BufReader::new(input);
+    for (line_num, line) in reader.lines().enumerate() {
+        // Read the line and trim it
+        let line: String = line.with_context(|| format!("line {}", line_num + 1))?;
+        let line = line.trim();
+
+        // Skip the line if it's a comment or empty
```

Comment:
Do these conditions come up in practice? What does coreutils do with these cases?

Reply:
I was following along with the behavior here: 

https://github.com/jtdowney/b2sum-rs/blob/master/src/main.rs#L127-L129

Reply:
`shasum` (on my mac at least) allows one trailing empty line, but errors on more than one consecutive empty line.

Reply:
If we can simplify things by demanding that the input format is exactly the same as the tool output, I'm all for simplifying.